storyscript/compiler/lowering/Lowering.py

```diff
@@ -459,18 +460,14 @@ def visit_assignment(self, node, block, parent):
                             Tree('string', [name])
                         ])
                     ]))
-                    if i + 1 == len(c.children):
-                        # for the last entry, we can recycle the existing node
-                        node.children[0] = n
-                        node.assignment_fragment.base_expression.children = \
-                            [val]
-                    else:
-                        # insert new fake line
-                        a = block.assignment_path(n, val, new_line)
-                        parent.children.insert(0, a)
+                    a = block.assignment_path(n, val, new_line, eq_tok=eq_tok)
+                    block.insert_node(a, name.line)
+                return True
         else:
             for c in node.children:
-                self.visit_assignment(c, block, parent=node)
+                did_do_insert = self.visit_assignment(c, block, parent=node)
```

Comment:
Maybe a more descriptive name like?

```suggestion
                performed_destructuring = self.visit_assignment(c, block, parent=node)
```